### ISSUE TITLE
Implement a larger set of basic "analysis" functions

### ISSUE THREAD

**Issue by eteq** [astropy/specutils#234]
There's a fairly comprehensive list of spectroscopic "primitive" analysis tools that was compiled as part of the specviz work: spacetelescope/specviz#425 - that overlaps with some of the items already in https://github.com/astropy/specutils/projects/1 - these should be merged at some point.

**Comment by eteq** [comment]
@brechmos-stsci - note that the the set of issues (#235, #236, #237, #238, #239, #240, #241, #242, #243, and  #244)  all probably fall under this "meta issue".

Two related items for consideration that go under all of these (partly discussed in spacetelescope/specviz#425):
* How do we want to handle "regions"?  See https://github.com/spacetelescope/specviz/issues/425#issuecomment-404229597 for my initial thoughts
* Do we want these all to live in `analysis`? (probably yes, like the example equivalent width that already exists).  But do we *also* want to require users to do ``from specutils.analysis import X``?  Or use the ``__all__`` + ``import *`` trick in ``__init__.py`` to let them instead do ``from specutils import X``?

**Comment by brechmos-stsci** [comment]
@eteq @crawfordsm For the fitting and measurement/analysis methods I would suggest the API `measure(spectrum, region=...)`

There are two reasons for this:
1.  It might be possible that the spectral measure in the region might need or benefit from information in other parts of the spectrum. I can't give an example now, but I could forsee that it might be possible.
2. You could potentially do things like `snrs = snr(spectrum, region=[region1, region2])` (akin to `np.percentile` etc). This might be helpful if you are tracking around several regions.  Essentially this just comes down to where the loop-over-regions lies but I suspect this will happen enough times that passing region in to the `measure` method would be beneficial.

**Comment by brechmos-stsci** [comment]
I personally think `from specutils.analysis import X` is more appropriate import.  It shows where `X` is coming from and hints at the use.  I would definitely advocate for this (rather than `from specutils import X`.

**Comment by eteq** [comment]
After some offline discussion with several randomly-chosen spectroscopists, I've come around to @brechmos-stsci 's perspective that measure(spectrum, region=...) is the better approach.

Also, out-of-band, @brechmos-stsci and I discussed whether the region definitions should be a distinct class (i.e. `SpectrumRegion`) vs a simple thing like a 2-tuple which uses `Spectrum1D` methods to translate into anything useful.  I think we came down on the latter, but will make an issue summarizing the to-do's in that area.
